fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/NavigationBar.kt

```diff
@@ -17,166 +16,320 @@ import androidx.compose.material.IconButton
+        )
+
+        ForwardButton(
+            onForwardButtonClick = onForwardButtonClick,
+            onForwardButtonLongPress = onForwardButtonLongPress,
+        )
+
+        HomeButton(
+            onHomeButtonClick = onHomeButtonClick,
+        )
+
+        TabsButton(
+            onTabsButtonClick = onTabsButtonClick,
+            tabCount = tabCount,
+        )
+
+        MenuButton(menuButton = menuButton)
     }
 }
 
 /**
- * Represents a navigation bar element.
+ * Top-level UI for displaying the navigation bar.
  *
- * @property iconId Resource ID of the icon that item should display.
- * @property descriptionResourceId Text used as a content description by accessibility services.
- * @property type Type of the item, defaults to [ItemType.STANDARD].
+ * @param onSearchButtonClick
+ * @param menuButton
+ * @param onTabsButtonClick
+ * @param isPrivateMode
+ * @param browserStore
  */
-data class ActionItem(
-    val iconId: Int,
-    val descriptionResourceId: Int,
-    val type: ItemType = ItemType.STANDARD,
-)
+@Composable
+fun HomeNavBar(
+    onSearchButtonClick: () -> Unit,
+    menuButton: MenuButton,
+    onTabsButtonClick: () -> Unit,
+    isPrivateMode: Boolean,
+    browserStore: BrowserStore,
+) {
+    val tabCount = browserStore.observeAsState(initialValue = 0) { browserState ->
+        if (isPrivateMode) {
+            browserState.privateTabs.size
+        } else {
+            browserState.normalTabs.size
+        }
+    }.value
 
-/**
- * Enumerates the types of items that can be used in a navigation bar.
- *
- * [STANDARD] - Represents a regular navigation item. Used for most navigation actions.
- * [TAB_COUNTER] - Represents a specialized item used to display a count, such as the number of open tabs in a browser.
- */
-enum class ItemType {
-    STANDARD, TAB_COUNTER, MENU
+    Row(
+        modifier = Modifier
+            .background(FirefoxTheme.colors.layer1)
+            .height(48.dp)
+            .fillMaxWidth()
+            .padding(horizontal = 16.dp),
+        horizontalArrangement = Arrangement.SpaceBetween,
+    ) {
+        BackButton(
+            onBackButtonClick = {},
+            onBackButtonLongPress = {},
+        )
+
+        ForwardButton(
+            onForwardButtonClick = {},
+            onForwardButtonLongPress = {},
+        )
+
+        SearchWebButton(
+            onSearchButtonClick = onSearchButtonClick,
+        )
+
+        TabsButton(
+            onTabsButtonClick = onTabsButtonClick,
+            tabCount = tabCount,
+        )
+
+        MenuButton(menuButton = menuButton)
+    }
 }
 
-/**
- * Provides a collection of navigation items used in the application's navigation bar.
- */
-object NavigationItems {
-    val home = ActionItem(
-        iconId = R.drawable.mozac_ic_home_24,
-        descriptionResourceId = R.string.browser_toolbar_home,
-    )
+@Composable
+private fun BackButton(
+    onBackButtonClick: () -> Unit,
+    onBackButtonLongPress: () -> Unit,
+) {
```

Comment:
for these button composables, we could technically consolidate them, but the juice likely isn't worth the squeeze

Reply:
so we keep it as it is, but for education purposes, what did you have in mind? I am not sure how consolidation would look like in this case